pkg/codegen/java/gen_program.go

```diff
@@ -146,7 +146,11 @@ func GenerateProgram(program *pcl.Program) (map[string][]byte, hcl.Diagnostics,
 
 	// Import Java-specific schema info.
 	// FIX ME: not sure what this is doing...
-	for _, p := range program.Packages() {
+	packages, err := program.PackageSnapshots()
```

Comment:
Since we're here, remove the FIX ME above with a brief comment on what it's doing? It looks like we need it somehow but I'm confused. I tried looking at the source behind ImportLanguages. Can't quite understand what's going on here. 

Sounds like the Java Importer implementation is mostly a no-op pass-through.  Do some languages hook into this interface to do something custom?

// Importer implements schema.Language for Java


CC @Zaid-Ajaj

Reply:
I think the language importer is supposed to load java-specific schema information such as property name overrides but I haven't seen a schema that actually makes use of it (and programgen doesn't either)